grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy

```diff
@@ -256,4 +253,67 @@ class PrepaymentInvoiceService {
         // to generate invoice item partially for the same invoice
         return relatedObject?.invoiceItems?.find { InvoiceItem it -> it.inverse && it.invoice == invoice }
     }
+
+    /**
+     * Bulk editing invoice items quantity on final invoice for prepaid POs.
+     * We are allowing to edit only the quantity here. After editing invoice items quantity we have to
+     * change quantity on the corresponding inverse item.
+     *
+     * */
+    void updateItemsQuantity(String invoiceId, List items) {
```

Comment:
It might be best to make this method more generic (ie call it updateItems instead of updateItemsQuantity). Right now we only allow you to update the quantity field but that could change in the future.